packages/addons/service/librespot/source/default.py

```diff
@@ -158,53 +155,70 @@ class Spotify():
     def __init__(self):
         self.headers = None
         self.expiration = time.time()
-        self.request = [
-            'https://accounts.spotify.com/api/token',
-            urllib.urlencode({'grant_type': 'client_credentials'}),
-            {'Authorization': 'Basic {}'.format(base64.b64encode(
-                '{}:{}'.format(SPOTIFY_ID, SPOTIFY_SECRET)))}
-        ]
 
     def getHeaders(self):
         if time.time() > self.expiration:
             log('token expired')
-            token = json.loads(urllib2.urlopen(
-                urllib2.Request(*self.request)).read())
+            token = requests.post(
+                url='https://accounts.spotify.com/api/token',
+                data={'grant_type': 'client_credentials'},
+                headers={
+                    'Authorization': 'Basic MTY5ZGY1NTMyZGVlNDdhNTk5MTNmODUyOGU4M2FlNzE6MWYzZDhiNTA3YmJlNGY2OGJlYjNhNDQ3MmU4YWQ0MTE='}
+            ).json()
+            print(token)
             log('new token expires in {} seconds'.format(token['expires_in']))
-            self.expiration = time.time() + float(token['expires_in']) - 60
+            self.expiration = time.time() + float(token['expires_in']) - 15
             self.headers = {
                 'Accept': 'application/json',
                 'Content-Type': 'application/json',
                 'Authorization': 'Bearer {}'.format(token['access_token'])
             }
 
-    def getTrack(self, track_id):
-        log('getting track')
+    def getEndpoint(self, url):
         try:
             self.getHeaders()
-            track = json.loads(urllib2.urlopen(urllib2.Request(
-                'https://api.spotify.com/v1/tracks/{}'.format(track_id), None,
-                self.headers)).read())
-        except Exception as e:
-            log('failed to get track from Spotify: {}'.format(e))
-            track = dict()
-        return Track(track)
+            return requests.get(
+                url=url,
+                headers=self.headers
+            ).json()
+        except e:
```

Comment:
```suggestion
        except Exception as e:
```
In this current form, `e` is undefined and will throw an unhandled `NameError` exception.

Reply:
Thank you for the review.
I fixed it.
Ready to merge.